### DISCUSSION TITLE
Clarification on terminology: 'Memory' vs 'Replay Buffer'

### DISCUSSION THREAD

**Discussion by jborbik** [Toni-SM/skrl#213] [Q&A] [answered]
Hello,

Thank you for providing such a well-documented and easy-to-follow RL framework!

I have a quick question regarding terminology: is there a specific reason why the replay buffers are referred to as "memories" in the code? From my understanding, the term "replay buffer" is more commonly used in the literature. I’d appreciate any clarification on this.

**Comment by Toni-SM** [comment, marked as answer]
Hi @Jendker 

`skrl` implemnents a **generic memory definition** that are not tied to the agent implementation and can be used for any role, such as rollout or replay buffers. The `skrl`'s memory is an empty shell, and the agents are in charge of defining the tensors according to their needs.

Depending on the agent, the memory can be used as:
* Replay buffer: large in size memories typically used by off-policy algorithms to storage environment transitions *(state, action, reward, next state, done)* with various sample strategies: random (implemented), prioritized, etc. 
* Rollout buffer: small in size memories typically used by on-policy algorithms to store batches collected by the current policy and that are discarded after use. Apart from the environment transition, other policy-specific data (such as policy *log_prob*) is collected.

**Reply by jborbik** [reply]
Thank you for your explanation! I appreciate that you took the time for it!

**Reply by Toni-SM** [reply]
Also, you can take a look at the way PPO (on-policy) and DDPG (off-policy) uses the same memory class for each particular case:

- PPO:
  - tensor creation: https://github.com/Toni-SM/skrl/blob/2bd3b7a25ba7f4813b0154909cc3e7965f4f0a57/skrl/agents/torch/ppo/ppo.py#L185-L197
  - sampling: https://github.com/Toni-SM/skrl/blob/2bd3b7a25ba7f4813b0154909cc3e7965f4f0a57/skrl/agents/torch/ppo/ppo.py#L378
- DDPG
  - tensor creation: https://github.com/Toni-SM/skrl/blob/2bd3b7a25ba7f4813b0154909cc3e7965f4f0a57/skrl/agents/torch/ddpg/ddpg.py#L179-L187
  - sampling: https://github.com/Toni-SM/skrl/blob/2bd3b7a25ba7f4813b0154909cc3e7965f4f0a57/skrl/agents/torch/ddpg/ddpg.py#L332-L333
